### ISSUE TITLE
cchost does not print an error when trying to access an invalid target RPC address

### ISSUE THREAD

**Issue by eddyashton** [microsoft/CCF#1486]
When launching a new node to join an existing network, we run `cchost ... join --target-rpc-address <address>`. If the address is invalid or unreachable, the joining node appears to hang and, with the default logging configuration, does not indicate that anything is wrong:

```
2020-08-06T12:29:39.506015Z        100 [info ] ../src/host/node_connections.h:161   | Listening for node-to-node on 127.98.160.202:37457
2020-08-06T12:29:39.506406Z        100 [info ] ../src/host/rpc_connections.h:75     | Listening for RPCs on 127.98.160.202:33653
2020-08-06T12:29:39.506562Z        100 [info ] ../src/host/main.cpp:608             | Creating new node - joining existing network at 127.1.2.3:1234
Azure Quote Provider: libdcap_quoteprov.so [ERROR]: Could not retreive environment variable for 'AZDCAP_DEBUG_LOG_LEVEL'
2020-08-06T12:29:39.553100Z        100 [info ] ../src/host/main.cpp:638             | Created new node
2020-08-06T12:29:39.553507Z -0.047 0   [info ] ../src/node/node_state.h:1071        | Node TLS connections now accepted
2020-08-06T12:29:39.563806Z -0.057 0   [info ] ../src/enclave/main.cpp:119          | Starting thread: 0
2020-08-06T12:29:39.563917Z -0.058 0   [info ] ../src/enclave/main.cpp:126          | All threads are ready!
```

Under the hood, the enclave is repeatedly sending join requests every time its timer expires, which attempts to create a connection via `tls::tls_connect`, which fails on the host side with all failure diagnostics at `DEBUG` and below.

We should make this failure more visible. At a minimum we should increase the verbosity of the RPC connection failure logging, and make it clear when this address appears invalid. We may also be able to specify an explicit behaviour for this connection which informs the _enclave_ of total connection failures.

**Comment by achamayou** [comment]
Slightly different but related: #733

**Comment by putyoutoosleep00** [comment]
Fuc you

**Comment by achamayou** [comment]
Offtopic comment containing inappropriate language deleted.

**Comment by letmaik** [comment]
I'm running into the same problem now when trying to figure out if JWT key refreshes were successful or not, which in the case of connection issues is impossible. I think the http client in the enclave should either have a second error callback, or have special values passed in the current callback, e.g. status=0.

**Comment by rengolin** [comment]
Just tried this now and IIUC, the logs are DEBUG, so won't show unless `VERBOSE_DEBUGGING=ON`.

Here's what I get:
```
[debug] ../src/enclave/rpc_sessions.h:318    | Creating a new client session inside the enclave: -2
[debug] ../src/host/rpc_connections.h:235    | rpc connect request from enclave -2
[debug] ../src/host/tcp.h:693                | uv_tcp_connect async retry: connection refused
[debug] ../src/host/tcp.h:545                | unable to connect: all resolved addresses failed: 127.161.12.218:46088
[debug] ../src/host/rpc_connections.h:34     | rpc connect failed -2
[debug] ../src/node/node_state.h:624         | Sending join request to 127.161.12.218:46088
[debug] ../src/host/rpc_connections.h:225    | rpc write from enclave -2: 326
[trace] ../src/enclave/tls_endpoint.h:595    | TLS Endpoint::send_cb() : Put 326 bytes
[trace] ../src/enclave/tls_endpoint.h:630    | TLS Endpoint::recv_cb() : WANTS_READ
[debug] ../src/enclave/rpc_sessions.h:298    | Closing a session inside the enclave: -2
[debug] ../src/host/rpc_connections.h:260    | rpc closed from enclave -2
```

Would it be enough to just raise the "unable to connect" log to INFO?

**Comment by achamayou** [comment]
Addressed by #3434
